sympy/polys/domains/mpelements.py

```diff
@@ -22,10 +22,7 @@ def _set_mpf(self, val):
         prec, rounding = self.context._prec_rounding
         tol = self.context.tol
 
-        if mpf_lt(mpf_abs(val, prec, rounding), tol):
-            self.__mpf__ = fzero
-        else:
-            self.__mpf__ = val
+        self.__mpf__ = val
```

Comment:
@smichr those are good points that will require further refactoring. I would delete any kind of truncation.

> Should any sort of truncation (perhaps in digits) be happening in this file?

I don't think so. But I didn't write this file, so I am not sure what the purpose was. @mattpap, what was the motivation behind truncating the numbers? Is there any reason to do so?